### PR TITLE
Graphql Implementation of Models

### LINKED ISSUE #45: API: Consider Using Graphql
Proposal to add [FEATURE] to to the site. 

**Feature Name** 
Consider using Graphql

**Functionality**
Graphql is great for getting all data needed in a single payload. Combined with an authorization library, it would be easy to layout authorization role based access, without us having to do a lot of custom controller work.

**Notes**

https://graphql.org/graphql-js/running-an-express-graphql-server/

### PR DESCRIPTION
**Pull request by crstamps2**
Fixes #45 

Couple of things this still needs.

1. Tests
2. Passport integration (though we should talk about following Apollo's opinions on that)
3. Wiring up the front end to use the new Graphql stuff.

I recommend squash merging this PR.

### CODE REVIEW

**Comment by crstamps2** [comment]
With this PR, you should be able to add the following to some kind of graphql playground:

```graphql
# Write your query or mutation here
mutation CreateWishForChild($child: ChildInput!, $details: String!, $type: Type!) {
  createWish(child: $child, details: $details, type: $type){
    id
    child {
      name
      hometown
      illness
      age
    }
    details
    type
  }
}


query Wishes{
  wishes {
    details
    type
    child {
      name
      age
      illness
      hometown
    }
  }
}```

File: api/src/resolvers/wish.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+module.exports = {
+  Query: {
+    wishes: async () => {
+      const wish = await Wish.find();
+      console.log(wish);
+      return wish;
+    },
+    wish: async (parent, { id }) => {
+      return await Wish.findById(id);
```
Comment (alex-hall) [inline api/src/resolvers/wish.js]: waiting here necessary?

**Comment by alex-hall** [comment]
So obviously tests are failing, and a bunch of tests were deleted. 

The GraphQL stuff itself looks good. 

What's the plan for Apollo on the front end? Is it necessary?

**Review by alex-hall — CHANGES_REQUESTED** [review]
See comments

**Comment by crstamps2** [comment]
Yeah I knew going into the PR that I was going to have some failing tests. Still trying to figure out how best to test the api.

Apollo on the front end is not necessary, react has some opinionated graphql concepts we can tap into.

File: api/src/resolvers/wish.js (hunk 1 above)
Comment (crstamps2) [reply api/src/resolvers/wish.js]: yes it is, because this is the database lookup.

**Comment by alex-hall** [comment]
I've used [https://github.com/visionmedia/supertest](Supertest) to integration tests express APIs in the past.

File: api/src/resolvers/wish.js (hunk 1 above)
Comment (fzondlo) [reply api/src/resolvers/wish.js]: I believe `await` is implied on a return from an async function. I don't think it's needed.

**Pull request closed without merging**